OpenRA.Mods.Common/Traits/BotModules/BotModuleLogic/BaseBuilderQueueManager.cs

```diff
@@ -51,6 +52,7 @@ class BaseBuilderQueueManager
 			failRetryTicks = baseBuilder.Info.StructureProductionResumeDelay;
 			minimumExcessPower = baseBuilder.Info.MinimumExcessPower;
 			this.resourceTypeIndices = resourceTypeIndices;
+			shouldCheckForWater = baseBuilder.Info.NavalProductionTypes.Any();
```

Comment:
You could also add a `WaterCheck.DontCheck` to the enumeration in `AIUtils`.

Reply:
good idea, I will try it.

Reply:
added and use it as watercheck